packages/server/src/errors/index.ts

```diff
@@ -8,3 +10,19 @@ export enum ApolloServerErrorCode {
   OPERATION_RESOLUTION_FAILURE = 'OPERATION_RESOLUTION_FAILURE',
   BAD_REQUEST = 'BAD_REQUEST',
 }
+
+/**
+ * unwrapResolverError is a useful helper function for `formatError` hooks.
+ * Errors thrown in resolvers are wrapped by graphql-js in a GraphQLError that
+ * adds context such as the `path` to the field in the operation. If you'd like
+ * to look directly at the original error thrown in the resolver (with whatever
+ * data is on that error object, but without fields like `path`), you can use
+ * this function. Note that other GraphQLErrors that contain `originalError`
+ * (like parse errors) are not unwrapped by this function.
+ */
+export function unwrapResolverError(error: unknown): unknown {
+  if (error instanceof GraphQLError && error.path && error.originalError) {
```

Comment:
Not sure I understand why we also need the `error.path` check here.

Unrelated: I didn't find anything interesting on this path but maybe you will. Pretty sure the incoming `unknown` makes any attempt at refining the return type pointless. Might only be useful in a context where you have some type info about the incoming `error` argument.
```ts
export function unwrapResolverError<T = unknown>(error: T): T | Error {
  if (error instanceof GraphQLError && error.path && error.originalError) {
    return error.originalError;
  }
  return error;
}
```

Reply:
@trevor-scheer Well, that's what makes this "unwrapResolverError" and not just "unwrapError".

Look at, say, SyntaxError — it's a GraphQLError defined in our code that wraps a GraphQLError received from graphql-js, adding `code` as an extension. I don't think it's helpful for a formatError to strip off our SyntaxError, is it?

Reply:
Got it, I was only seeing this as a type-level refinement and didn't see why we needed it to access `originalError`.